common/types.go

```diff
@@ -179,6 +179,16 @@ func (h Hash) Value() (driver.Value, error) {
 	return h[:], nil
 }
 
+// IsZero checks if the Hash is a zero value
+func (h Hash) IsZero() bool {
+	for _, b := range h {
+		if b != 0 {
+			return false
```

Comment:
I don't think this implementation is fastest, I suspect the fastest is to just compare against `==(common.Hash{})`. For contender, I'd guess that looping and `OR`-ing the 32 bytes into one and comparing the sum against zero-byte would be number two (branchless execution).

Reply:
Thanks for the heads up.

Just do a benchmark, using the comparison is faster.

I will close this pr.